OpenSim/Analyses/StaticOptimization.cpp

```diff
@@ -589,8 +587,18 @@ int StaticOptimization::begin(const SimTK::State& s )
         _forceReporter->begin(sWorkingCopy);
         _forceReporter->updForceStorage().reset();
 
+        // Set initial guess to lowest activation
         _parameters.resize(_modelWorkingCopy->getNumControls());
-        _parameters = 0;
+        const Set<Actuator>& fs = _modelWorkingCopy->getActuators();
+        for(int i=0,j=0;i<fs.getSize();i++) {
+            ScalarActuator* act = dynamic_cast<ScalarActuator*>(&fs.get(i));
+            if (act) {
+                if (act->getMinControl() != -INFINITY)
+                    _parameters[j++] = act->getMaxControl();
```

Comment:
Do you mean act ->get**Min**Control()?

Reply:
Actually, the error is in the if. Unfortunately, contrary to what is previously done, the initial guess has on the first frame has to be maxControl. The reason is that I linearize between 0 (actually MinControl) and value from previous frame. If we set the previous frame to be minimal, it is a singularity. Setting it to max solves to the same value as the previous linear version : the problem being quadratic when p=2 and therefore the solver finds a good optimal solution even though I initialize at max.